src/amberscript/parser.cc

```diff
@@ -1301,20 +1301,55 @@ Result Parser::ParsePipelineVertexData(Pipeline* pipeline) {
   const uint32_t location = token->AsUint32();
 
   InputRate rate = InputRate::kVertex;
+  uint32_t offset = 0;
+  Format* format = buffer->GetFormat();
+  uint32_t stride = format->SizeInBytes();
```

Comment:
This is kind of subtle but I think if somebody overrides the format, but not the stride, then the stride should be derived from the overridden format, not the format of the original buffer. I guess you can check if the stride was explicitly set after the loop (I.e. It is non-zero? Although I guess we should not allow an explicit 0 stride?) and, if not, set it based on the format. And I guess add a test for this as well.

Reply:
done